test/unit/model/DataStore/MetaDataDeliverySyncTaskTest.php

```diff
@@ -0,0 +1,116 @@
+use taoQtiTest_models_classes_QtiTestService;
+
+class MetaDataDeliverySyncTaskTest extends TestCase
+{
+    use OntologyMockTrait;
+
+    /** @var QueueDispatcher|\PHPUnit\Framework\MockObject\MockObject */
+    private $queueDispatcher;
+
+    /** @var PersistDataService|\PHPUnit\Framework\MockObject\MockObject */
+    private $persistDataService;
+
+    /** @var \PHPUnit\Framework\MockObject\MockObject|taoQtiTest_models_classes_QtiTestService */
+    private $qtiTestService;
+
+    /** @var ResourceJsonMetadataCompiler|\PHPUnit\Framework\MockObject\MockObject */
+    private $resourceJsonMetadataCompiler;
+
+    /** @var \Zend\ServiceManager\ServiceLocatorInterface */
+    private $serviceLocator;
+
+    /** @var core_kernel_persistence_smoothsql_SmoothModel */
+    private $ontology;
+
+    /** @var MetaDataDeliverySyncTask */
+    private $subject;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->queueDispatcher = $this->createMock(QueueDispatcher::class);
+        $this->persistDataService = $this->createMock(PersistDataService::class);
+        $this->qtiTestService = $this->createMock(taoQtiTest_models_classes_QtiTestService::class);
+        $this->resourceJsonMetadataCompiler = $this->createMock(ResourceJsonMetadataCompiler::class);
+        $this->ontology = $this->getOntologyMock();
+
+        $this->serviceLocator = $this->getServiceLocatorMock([
+            QueueDispatcher::SERVICE_ID => $this->queueDispatcher,
+            PersistDataService::class => $this->persistDataService,
+            taoQtiTest_models_classes_QtiTestService::class => $this->qtiTestService,
+            ResourceJsonMetadataCompiler::SERVICE_ID => $this->resourceJsonMetadataCompiler,
+            Ontology::SERVICE_ID => $this->ontology
+        ]);
+
+        $this->subject = new MetaDataDeliverySyncTask();
+    }
+
+    public function testJsonSerialize(): void
+    {
+        $this->assertEquals(
+            MetaDataDeliverySyncTask::class,
+            $this->subject->jsonSerialize()
+        );
+    }
+
+    public function testInvoke()
+    {
+        $this->persistDataService->method('persist');
+        $this->qtiTestService->method('getItems')->willReturn([]);
+        $this->queueDispatcher->method('createTask')->willReturn(true);
+        $this->resourceJsonMetadataCompiler->method('compile')->willReturn([]);
+        $this->subject->setServiceLocator($this->serviceLocator);
+
+        $class = $this->ontology->getClass('http://tao.tld/bogusUri');
+        $mockDelivery = $class->createInstance('Bogus');
+        $mockTest = $class->createInstance('TestBogus');
+        $mockDelivery->setPropertyValue(
+            $this->ontology->getProperty(DeliveryAssemblyService::PROPERTY_ORIGIN),
+            $mockTest
+        );
+
+        $param = [
+            'deliveryId' => $mockDelivery->getUri(),
+            'max_tries' => 1,
+            'count' => 0
+        ];
+        $subject = $this->subject;
```

Comment:
```suggestion
        $result = $this->subject->__invoke($param);
```

Reply:
it is still the subject that i am testing and not a result

Reply:
Yes. I understand. I wanted to point that it's more readable when you explicitly call invoke method in a test. It's optional thought.